src-electron/generator/helper-zcl.js

```diff
@@ -2693,6 +2698,261 @@ function if_compare(leftValue, rightValue, options) {
   }
 }
 
+/**
+ *
+ * @param {*} type
+ * @param {*} dataType
+ * @param {*} packageIds
+ * @param {*} context
+ * @returns The size and sign of a zcl data type
+ */
+async function zcl_data_type_size_and_sign(
+  type,
+  dataType,
+  packageIds,
+  context
+) {
+  let result = 0
+  let isSigned = false
+  if (dataType.discriminatorName.toLowerCase() == dbEnum.zclType.bitmap) {
+    let bitmap = await queryZcl.selectBitmapByName(
+      context.global.db,
+      packageIds,
+      dataType.name
+    )
+    result = bitmap.size
+  } else if (dataType.discriminatorName.toLowerCase() == dbEnum.zclType.enum) {
+    let en = await queryZcl.selectEnumByName(
+      context.global.db,
+      dataType.name,
+      packageIds
+    )
+    result = en.size
+  } else if (
+    dataType.discriminatorName.toLowerCase() == dbEnum.zclType.number
+  ) {
+    let number = await queryZcl.selectNumberByName(
+      context.global.db,
+      packageIds,
+      dataType.name
+    )
+    isSigned = number.isSigned
+    result = number.size
+  } else {
+    env.logWarning(type + ' is a complex type and size could not be determined')
+  }
+  return { size: result, isSigned: isSigned }
+}
+
+/**
+ * Available Options:
+ * - asUpperCamelCase: 0/1 to return cpp data type starting with lower/upper
+ * camel case
+ * @param {*} type
+ * @param {*} dataType
+ * @param {*} hash
+ * @param {*} context
+ * @returns The corresponding cpp data type for a zcl data type.
+ */
+async function as_underlying_cpp_zcl_type(type, dataType, hash, context) {
+  let res = await new Promise((resolve, reject) => {
+    let asUpperCamelCase =
+      hash && hash.asUpperCamelCase ? hash.asUpperCamelCase : false
+    if (dataType.discriminatorName.toLowerCase() == dbEnum.zclType.string) {
+      if (octetStringTypes.includes(type.toUpperCase())) {
+        resolve('OctetString')
```

Comment:
I'm not sure I follow this.  There's no OctetString C++ type in the Matter sdk.  Or CharString.

Reply:
Sorry this was objective C. Fixed this